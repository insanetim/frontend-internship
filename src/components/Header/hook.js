import { useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useLocation, useNavigate } from 'react-router-dom'

import { logOut } from 'src/state/session/actions'
import { accountSelector } from 'src/state/session/selectors'
import { redirect } from 'src/utils'

const useContainer = () => {
  const dispatch = useDispatch()
  const account = useSelector(accountSelector)
  const location = useLocation()
  const navigate = useNavigate()
  const cb = useCallback(redirect('/login', location, navigate), [location, navigate])

  const handleLogout = () => {
    dispatch(logOut(cb))
  }

  return { account, handleLogout }
}

export default useContainer
