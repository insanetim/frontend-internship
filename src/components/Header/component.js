import React from 'react'
import { Typography, Row, Col, Avatar, Dropdown, Menu, Layout } from 'antd'
import { CaretDownOutlined, UserOutlined } from '@ant-design/icons'
import { Link, useLocation } from 'react-router-dom'
import { isEmpty } from 'lodash'

import { APP_NAME } from 'src/constants'
import { useContainer } from './hook'

const Header = () => {
  const { account, handleLogout } = useContainer()
  const location = useLocation()

  const items = [
    { label: <Link to={`/${location.search}`}>Dashboard</Link> },
    { type: 'divider' },
    { label: <Link to='/lists'>My Lists</Link> },
    { label: <Link to='/watchlist'>Watchlist</Link> },
    { label: <Link to='/favorites'>Favorites</Link> },
    { type: 'divider' },
    { label: 'Logout', onClick: () => handleLogout() }
  ]
  const menu = <Menu items={items} />

  return (
    <Layout.Header>
      <Row
        type='flex'
        justify='space-between'
      >
        <Col>
          <Typography.Text>{APP_NAME}</Typography.Text>
        </Col>
        <Col>
          <Dropdown overlay={menu}>
            <Typography.Text>
              {isEmpty(account) ? (
                <Avatar icon={<UserOutlined />} />
              ) : (
                <Avatar src={`https://www.gravatar.com/avatar/${account.avatar.gravatar.hash}`} />
              )}{' '}
              <span className='hide-sm-down'>{isEmpty(account) ? 'Username' : account.username}</span>{' '}
              <CaretDownOutlined />
            </Typography.Text>
          </Dropdown>
        </Col>
      </Row>
    </Layout.Header>
  )
}

export default Header
