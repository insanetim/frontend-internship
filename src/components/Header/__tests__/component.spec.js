/* eslint-disable react/jsx-no-useless-fragment */
import { shallow } from 'enzyme'

import Header from '../component'

jest.mock('react-router-dom', () => ({
  Link: () => <></>
}))

const mockedHookData = {
  account: {
    avatar: {
      gravatar: {
        hash: 'test/avatar'
      }
    },
    username: 'test/username'
  },
  handleLogout: jest.fn()
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

describe('CreateListModal component tests', () => {
  let component = shallow(<Header />)

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('matches snapshot with empty account', () => {
    mockedHookData.account = {}
    component = shallow(<Header />)

    expect(component).toMatchSnapshot()
  })
})
