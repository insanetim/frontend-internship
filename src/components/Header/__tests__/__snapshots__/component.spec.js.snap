// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateListModal component tests matches snapshot 1`] = `
<Header>
  <Row
    justify="space-between"
  >
    <Col>
      <ForwardRef(Text)>
        THE MOVIE DB
      </ForwardRef(Text)>
    </Col>
    <Col>
      <Dropdown
        overlay={
          <Menu
            items={
              [
                {
                  "label": <Link
                    to="/"
                  >
                    Dashboard
                  </Link>,
                },
                {
                  "type": "divider",
                },
                {
                  "label": <Link
                    to="/lists"
                  >
                    My Lists
                  </Link>,
                },
                {
                  "label": <Link
                    to="/watchlist"
                  >
                    Watchlist
                  </Link>,
                },
                {
                  "label": <Link
                    to="/favorites"
                  >
                    Favorites
                  </Link>,
                },
                {
                  "type": "divider",
                },
                {
                  "label": "Logout",
                  "onClick": [MockFunction],
                },
              ]
            }
          />
        }
      >
        <ForwardRef(Text)>
          <Avatar
            src="https://www.gravatar.com/avatar/test/avatar"
          />
           
          <span
            className="hide-sm-down"
          >
            test/username
          </span>
           
          <ForwardRef(CaretDownOutlined) />
        </ForwardRef(Text)>
      </Dropdown>
    </Col>
  </Row>
</Header>
`;

exports[`CreateListModal component tests matches snapshot with empty account 1`] = `
<Header>
  <Row
    justify="space-between"
  >
    <Col>
      <ForwardRef(Text)>
        THE MOVIE DB
      </ForwardRef(Text)>
    </Col>
    <Col>
      <Dropdown
        overlay={
          <Menu
            items={
              [
                {
                  "label": <Link
                    to="/"
                  >
                    Dashboard
                  </Link>,
                },
                {
                  "type": "divider",
                },
                {
                  "label": <Link
                    to="/lists"
                  >
                    My Lists
                  </Link>,
                },
                {
                  "label": <Link
                    to="/watchlist"
                  >
                    Watchlist
                  </Link>,
                },
                {
                  "label": <Link
                    to="/favorites"
                  >
                    Favorites
                  </Link>,
                },
                {
                  "type": "divider",
                },
                {
                  "label": "Logout",
                  "onClick": [MockFunction],
                },
              ]
            }
          />
        }
      >
        <ForwardRef(Text)>
          <Avatar
            icon={<ForwardRef(UserOutlined) />}
          />
           
          <span
            className="hide-sm-down"
          >
            Username
          </span>
           
          <ForwardRef(CaretDownOutlined) />
        </ForwardRef(Text)>
      </Dropdown>
    </Col>
  </Row>
</Header>
`;
